src/components/DatePicker/index.ios.js

```diff
@@ -0,0 +1,133 @@
+    }
+
+    /**
+     * Reset the date spinner to the initial value
+     */
+    reset() {
+        this.setState({selectedDate: this.initialValue});
+    }
+
+    /**
+     * Accept the current spinner changes, close the spinner and propagate the change
+     * to the parent component (props.onChange)
+     */
+    selectDate() {
+        this.setState({isPickerVisible: false});
+        this.props.onChange(this.state.selectedDate);
+    }
+
+    /**
+     * @param {Event} event
+     * @param {Date} selectedDate
+     */
+    updateLocalDate(event, selectedDate) {
+        this.setState({selectedDate});
+    }
+
+    render() {
+        const {
+            value,
+            label,
+            placeholder,
+            hasError,
+            errorText,
+            translateX,
+            containerStyles,
+            disabled,
+        } = this.props;
+
+        const dateAsText = value ? moment(value).format(CONST.DATE.MOMENT_FORMAT_STRING) : '';
+
+        return (
+            <>
+                <ExpensiTextInput
+                    label={label}
+                    value={dateAsText}
+                    placeholder={placeholder}
+                    hasError={hasError}
+                    errorText={errorText}
+                    containerStyles={containerStyles}
+                    translateX={translateX}
+                    onPress={this.showPicker}
+                    editable={false}
+                    disabled={disabled}
+                />
+                <Popover
+                    isVisible={this.state.isPickerVisible}
+                    onClose={this.selectDate}
+                >
+                    <View style={[
+                        styles.flexRow,
+                        styles.justifyContentBetween,
+                        styles.borderBottom,
+                        styles.pb1,
+                        styles.ph4,
+                    ]}
+                    >
+                        <Button
+                            title={this.props.translate('common.reset')}
+                            color={themeColors.textError}
+                            onPress={this.reset}
+                        />
+                        <Button
+                            title={this.props.translate('common.done')}
+                            color={themeColors.link}
+                            onPress={this.selectDate}
+                        />
+                    </View>
+                    <RNDatePicker
+                        value={this.state.selectedDate}
+                        mode="date"
+                        display="spinner"
+                        onChange={this.updateLocalDate}
+                        locale={this.props.preferredLocale}
+                    />
+                </Popover>
+            </>
+        );
+    }
+}
+
+Datepicker.propTypes = datepickerPropTypes;
+Datepicker.defaultProps = defaultProps;
+
+export default withLocalize(Datepicker);
```

Comment:
NAB but maybe useful to mention _somewhere_ that iOS is the only platform where we have to manage localization ourselves, and on all others it's controlled by the OS

Reply:
I'll open a separate request to put a note about this

We're localizing the "Reset" and "Done" button that we put on our own popup. 
`RNDatepicker` would be shown in the system locale. I'm passing the `locale` to `RNDatepicker` because it will be weird to have the "Done" and "Reset" buttons in one locale and the picker months in another - it can happen

On Android the buttons and the modal are handled by the system so they are always in the system locale